### PR TITLE
Docs changes for adding IPTABLES_BACKEND to EGW deployment.

### PR DESCRIPTION
**Pull request by sridhartigera**
Added a new environment variable IPTABLES_BACKEND to the EGW deployment, 
Product Version(s):

Issue:

Link to docs preview:

- [ ] Deploy preview inspected wherever changes were made
- [ ] Build completed successfully
- [ ] Tests passed

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

### CODE REVIEW

**Review by girishshankaran — APPROVED** [review]
LGTM

**Comment by ctauchen** [comment]
@sridhartigera Can you provide more context for this? A link to an issue? Is this associated with a release?

**Comment by sridhartigera** [comment]
> @sridhartigera Can you provide more context for this? A link to an issue? Is this associated with a release?

@ctauchen Link to the issue https://tigera.atlassian.net/browse/CI-1226. This should go in as part of the next patch release for 3.15.2

File: calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx [hunk 1]
```diff
@@ -770,6 +770,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+         # Optional: Used to specify IP tables backend (legacy or nft). If not specified, egress gateway will auto-detect.
+         # If both nft and legacy are supported, nft will be used.
+        - name: IPTABLES_BACKEND
+          value: "nft"
```
Comment (fasaxc) [inline calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx]: We shouldn't default to `nft` in the docs.  Users will copy-paste without checking every flag is correct.  Better to have optional parameters commented out in some fashion, I think.

**Comment by fasaxc** [comment]
If you address my comment, any reason this couldn't go in without a release?  The env var has existed for a long time, we just didn't document it.

File: calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx (hunk 1 above)
Comment (fasaxc) [reply calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx]: Worth saying "for example on RHEL 9" or something for when you might want to use nft.

File: calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx [hunk 2]
```diff
@@ -770,6 +770,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+        # Optional: Used to specify IP tables backend (legacy or nft).
+        # For example on RHEL9, nft can be used.
+        #- name: IPTABLES_BACKEND
+          #value: "legacy"
```
Comment (fasaxc) [inline calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-aws.mdx]:
```suggestion
        #  value: "legacy"
```
Think this'd look cleaner

File: calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-on-prem.mdx [hunk 3]
```diff
@@ -300,6 +300,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+        # Optional: Used to specify IP tables backend (legacy or nft).
+        # For example on RHEL9, nft can be used.
+        #- name: IPTABLES_BACKEND
+          #value: "legacy"
```
Comment (fasaxc) [inline calico-enterprise_versioned_docs/version-3.15/networking/egress/egress-gateway-on-prem.mdx]:
```suggestion
        #  value: "legacy"
```

**Review by fasaxc — DISMISSED** [review]
One nit but I don't need to re-review.

**Pull request merged**